packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -177,16 +176,7 @@ export class RunningSummarizer implements IDisposable {
 
     public handleSystemOp(op: ISequencedDocumentMessage) {
         switch (op.type) {
-            case MessageType.ClientLeave: {
```

Comment:
Note RE this change: I tried to understand it better by replacing it with assert and running through various tests, including stress tests - I was never been able to hit it. I think it's due to a fact that before latest change with noops, client may have been removed from room (receive leave op for itself) on 5 min idle interval, and in such state connection would not be torn down but would downgrade to "read" without runtime layers understanding it.
This no longer can happen with more frequent noops.
If we get into same situation again, I'd argue the better fix is to ensure that all layers are aware that client lost "write" connection by cycling through synthetized disconnect/reconnect transitions. That way summarizer would know that it's cancelled. Just directly close connection if DeltaManager_reconnectMode !== ReconnectMode.Enabled.

Reply:
How will we know if we get into this situation? Should we keep the assert that you tried with or add telemetry?

Reply:
I do not think this code should care about it. I'll open an issue to consider these transitions at loader level - I think it's fine (for starters) to log an error that client should never receive its own leave message (I'll try if it's the case) and connection should be closed instead first. 
I'll add it to already opened https://github.com/microsoft/FluidFramework/issues/7180 that tracks it from different angle (as observed by ordering service).